include/deal2lkit/kinsol_interface.h

```diff
@@ -0,0 +1,184 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2015 by the deal2lkit authors
+//
+//    This file is part of the deal2lkit library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef _d2k_kinsol_interface_h
+#define _d2k_kinsol_interface_h
+
+#include <deal2lkit/config.h>
+#include <deal.II/base/config.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/conditional_ostream.h>
+
+#ifdef D2K_WITH_SUNDIALS
+// For time integration.
+#include <ida/ida.h>
+#include <ida/ida_spils.h>
+#include <ida/ida_spgmr.h>
+#include <ida/ida_spbcgs.h>
+#include <ida/ida_sptfqmr.h>
+#include <nvector/nvector_serial.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include <deal2lkit/sundials_interface.h>
+#include <deal2lkit/parameter_acceptor.h>
+
+#ifdef DEAL_II_WITH_MPI
+#include "mpi.h"
+#endif
+
+#include <kinsol/kinsol_impl.h>
+
+using namespace dealii;
+
+
+D2K_NAMESPACE_OPEN
+
+/** Interface to \sundials KINSOL library.
+
+\dk features an interface for the SUite of Nonlinear and
+DIfferential/ALgebraic equation Solvers (\sundials).
+
+The class KINSOLInterface is a wrapper to the KINSOL solver in \sundials. This is a general-purpose
+nonlinear system solver based on Newton-Krylov solver technology.
+Developed with KINSOL v.2.8.0.
+
+Citing from the \sundials documentation:
```

Comment:
remove these three lines, or actually add something instead of the blah blah blah... :)